### PR TITLE
feat(search): no results view

### PR DESCRIPTION
**Pull request by Aallam**

### CODE REVIEW

File: lib/ui/screens/products/components/paged_hits_grid_view.dart [hunk 1]
```diff
@@ -5,11 +5,15 @@ import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
...
-      {Key? key, required this.pagingController, this.onHitClick})
+      {Key? key,
+      required this.pagingController,
+      this.onHitClick,
+      this.onItemsFound})
       : super(key: key);
 
   final PagingController<int, Product> pagingController;
   final Function(String)? onHitClick;
+  final WidgetBuilder? onItemsFound;
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_grid_view.dart:16]: I suppose there is a typo here :) 
```suggestion
  final WidgetBuilder? noItemsFound;
```

File: lib/ui/screens/products/components/paged_hits_grid_view.dart [hunk 2]
```diff
@@ -23,6 +27,7 @@ class PagedHitsGridView extends StatelessWidget {
         crossAxisCount: 2,
       ),
       builderDelegate: PagedChildBuilderDelegate<Product>(
+        noItemsFoundIndicatorBuilder: onItemsFound,
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_grid_view.dart:30]:
```suggestion
        noItemsFoundIndicatorBuilder: noItemsFound,
```

File: lib/ui/screens/products/components/paged_hits_grid_view.dart [hunk 3]
```diff
@@ -5,11 +5,15 @@ import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
 
 class PagedHitsGridView extends StatelessWidget {
   const PagedHitsGridView(
-      {Key? key, required this.pagingController, this.onHitClick})
+      {Key? key,
+      required this.pagingController,
+      this.onHitClick,
+      this.onItemsFound})
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_grid_view.dart:11]:
```suggestion
      this.noItemsFound})
```

File: lib/ui/screens/products/components/paged_hits_list_view.dart [hunk 4]
```diff
@@ -5,11 +5,15 @@ import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
 
 class PagedHitsListView extends StatelessWidget {
   const PagedHitsListView(
-      {Key? key, required this.pagingController, this.onHitClick})
+      {Key? key,
+      required this.pagingController,
+      this.onHitClick,
+      this.onItemsFound})
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_list_view.dart:11]:
```suggestion
      this.noItemsFound})
```

File: lib/ui/screens/products/components/paged_hits_list_view.dart [hunk 5]
```diff
@@ -5,11 +5,15 @@ import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
...
       : super(key: key);
 
   final PagingController<int, Product> pagingController;
   final Function(String)? onHitClick;
+  final WidgetBuilder? onItemsFound;
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_list_view.dart:16]:
```suggestion
  final WidgetBuilder? noItemsFound;
```

File: lib/ui/screens/products/components/paged_hits_list_view.dart [hunk 6]
```diff
@@ -18,6 +22,7 @@ class PagedHitsListView extends StatelessWidget {
       pagingController: pagingController,
       separatorBuilder: (context, index) => const SizedBox(height: 10),
       builderDelegate: PagedChildBuilderDelegate<Product>(
+        noItemsFoundIndicatorBuilder: onItemsFound,
```
Comment (VladislavFitz) [inline lib/ui/screens/products/components/paged_hits_list_view.dart:25]:
```suggestion
        noItemsFoundIndicatorBuilder: noItemsFound,
```

File: lib/ui/screens/products/search_results_screen.dart [hunk 7]
```diff
@@ -58,91 +58,98 @@ class _SearchResultsScreen extends State<SearchResultsScreen> {
...
+  }
+
   Widget _hitsDisplay() {
     switch (_display) {
       case _HitsDisplay.list:
         return PagedHitsListView(
             pagingController: _pagingController,
-            onHitClick: (objectID) => _presentProductPage(context, objectID));
+            onHitClick: (objectID) => _presentProductPage(context, objectID),
+            onItemsFound: _noResults);
```
Comment (VladislavFitz) [inline lib/ui/screens/products/search_results_screen.dart:147]:
```suggestion
            noItemsFound: _noResults);
```

File: lib/ui/screens/products/search_results_screen.dart [hunk 8]
```diff
@@ -58,91 +58,98 @@ class _SearchResultsScreen extends State<SearchResultsScreen> {
...
       case _HitsDisplay.grid:
         return PagedHitsGridView(
             pagingController: _pagingController,
-            onHitClick: (objectID) => _presentProductPage(context, objectID));
+            onHitClick: (objectID) => _presentProductPage(context, objectID),
+            onItemsFound: _noResults);
```
Comment (VladislavFitz) [inline lib/ui/screens/products/search_results_screen.dart:152]:
```suggestion
            noItemsFound: _noResults);
```

**Review by VladislavFitz — APPROVED** [review]

File: lib/ui/screens/products/components/paged_hits_grid_view.dart (hunk 1 above)
Comment (Aallam) [reply lib/ui/screens/products/components/paged_hits_grid_view.dart:16]: Yes, nice catch!

**Pull request merged**